client/lib/domains/dns/index.js

```diff
@@ -45,8 +46,8 @@ function isValidDomain( name ) {
 	return /^([a-z0-9-_]{1,63}\.)*[a-z0-9-]{1,63}\.[a-z]{2,63}$/i.test( name );
 }
 
-function isValidName( name, type ) {
-	if ( isNameEmptyAndItsAllowed( name, type ) ) {
+function isValidName( name, type, domainName ) {
+	if ( isRootDomainAndItsAllowed( name, type, domainName ) ) {
```

Comment:
It's important to check if the input is a root domain _and_ it can be a root domain for the give type. That's why I wanted it to be one function (plus, it's used in two places in that file). But you are right - your version makes it clearer what's what. I'll update the PR :+1: